mod/gc_mobile_api/models/blog.php

```diff
@@ -244,3 +264,154 @@ function get_blogposts_by_owner($user, $limit, $offset, $lang, $target)
+		 'description2' => '',
+		 'description3' => '',
+		 'status' => $status,
+		 'access_id' => $access,
+		 'comments_on' => $comments,
+		 'excerpt' => JSON_encode($excerpts),
+		 'excerpt2' => '',
+		 'excerpt3' => '',
+		 'tags' => '',
+		 'publication_date' => '',
+		 'expiration_date' => '',
+		 'show_owner' => 'no'
+	 );
+
+	 //return $values;
+	 //return elgg_echo("test: pre blog creation");
+	 //Create blog
+	 $blog = new ElggBlog();
+	 $blog->subtype = 'blog';
+	 $blog->container_guid = $container_guid;
+	 $new_post = TRUE;
+
+	 $old_status = $blog->status;
+
+	 // assign values to the entity, stopping on error.
+	 if (!$error) {
+		 foreach ($values as $name => $value) {
+			 if (($name != 'title2') && ($name != 'description2') &&  ($name != 'excerpt2')){ // remove input 2 in metastring table
+			 $blog->$name = $value;
+			 }
+		 }
+	 }
+
+	 if (!$error){
+		 if ($blog->save()){
+
+				 $icon_file = get_resized_image_from_uploaded_file("icon", 100, 100);
+				 $icon_sizes = elgg_get_config("icon_sizes");
+
+				 if (!empty($icon_file) && !empty($icon_sizes)) {
+					 // create icon
+					 $prefix = "blogs/" . $blog->getGUID();
+
+					 $fh = new ElggFile();
+					 $fh->owner_guid = $blog->getOwnerGUID();
+
+					 foreach ($icon_sizes as $icon_name => $icon_info) {
+						 $icon_file = get_resized_image_from_uploaded_file("icon", $icon_info["w"], $icon_info["h"], $icon_info["square"], $icon_info["upscale"]);
+						 if (!empty($icon_file)) {
+							 $fh->setFilename($prefix . $icon_name . ".jpg");
+
+							 if ($fh->open("write")) {
+								 $fh->write($icon_file);
+								 $fh->close();
+							 }
+						 }
+					 }
+
+					 $blog->icontime = time();
+			 }
+			 // no longer a brand new post.
+			 $blog->deleteMetadata('new_post');
+
+			 $status = $blog->status;
+				 // add to river if changing status or published, regardless of new post
+				 // because we remove it for drafts.
+				 if (($new_post || $old_status == 'draft' ||  $old_status == 'published') && $status == 'published') {
+					 elgg_create_river_item(array(
+						 'view' => 'river/object/blog/create',
+						 'action_type' => 'create',
+						 'subject_guid' => $blog->owner_guid,
+						 'object_guid' => $blog->getGUID(),
+					 ));
+					 // we only want notifications sent when post published
+					 elgg_trigger_event('publish', 'object', $blog);
+
+					 // reset the creation time for posts that move from draft to published
+					 if ($guid) {
+						 $blog->time_created = time();
+						 $blog->save();
+					 }
+				 } elseif ($old_status == 'published' && $status == 'draft') {
+					 elgg_delete_river(array(
+						 'object_guid' => $blog->guid,
+						 'action_type' => 'create',
+					 ));
+				 }
+				 if ($blog->status == 'published' || $save == false) {
+					 return ($blog->getURL());
+				 } else {
+					 return ("blog/edit/$blog->guid");
+				 }
+
+		 } else {
+			 return elgg_echo('blog:error:cannot_save');
+		 }
+	 }
+
+	 return "end, unknown result";
```

Comment:
Should probably remove this before pushing to production, other than that - looks good!